### ISSUE TITLE
[DOXIA-734] XHTML 1.0 elements/attributes which are obsolete in XHTML5 no longer detected by XdocParser/FmlParser

### ISSUE THREAD

**Issue by jira-importer** [apache/maven-doxia#861]
**[Konrad Windszus](https://issues.apache.org/jira/secure/ViewProfile.jspa?name=kwin)** opened **[DOXIA-734](https://issues.apache.org/jira/browse/DOXIA-734?redirect=false)** and commented

Currently only `<a id=...>` is translated into a proper `Sink.anchor(...)` call.
The obsolete `<a name=...>` is currently not calling an according `Sink.anchor(...)` method but may be still used in ancient markups which derive from the XHtml5BaseParser.

For example XDoc and FML both stem from the time where https://www.w3.org/TR/xhtml1/ was the most recent XHTML spec and therefore support `a name` (according to https://maven.apache.org/doxia/doxia/doxia-modules/doxia-module-xdoc/xsddoc/http___maven.apache.org_XDOC_2.0/element/a.html and https://maven.apache.org/doxia/doxia/doxia-modules/doxia-module-fml/xsddoc/http___maven.apache.org_FML_1.0.1/element/a.html)

---

**Issue Links:**
- [DOXIA-668](https://issues.apache.org/jira/browse/DOXIA-668) Remove all obsolete attributes in HTML5
 (_**"is caused by"**_)

**Remote Links:**
- [GitHub Pull Request #204
](https://github.com/apache/maven-doxia/pull/204)

**Backported to:** [2.0.0-M10](https://github.com/apache/maven-doxia/milestone/42?closed=1)

**Comment by jira-importer** [comment]
**[Konrad Windszus](https://issues.apache.org/jira/secure/ViewProfile.jspa?name=kwin)** commented

This is a regression of DOXIA-682, as both parsers rely on Xhtml5Parser for the anchor element.

**Comment by jira-importer** [comment]
**[Konrad Windszus](https://issues.apache.org/jira/secure/ViewProfile.jspa?name=kwin)** commented

[Michael Osipov](https://issues.apache.org/jira/secure/ViewProfile.jspa?name=michael-o) I think basically all changes from https://issues.apache.org/jira/browse/DOXIA-668 need to be reverted for FML/APT because those languages don't derive from XHTML5 but rather XHTML1.

**Comment by jira-importer** [comment]
**[Konrad Windszus](https://issues.apache.org/jira/secure/ViewProfile.jspa?name=kwin)** commented

For me both FML and XDoc are legacy, therefore I see no reason to come up with a new (backwards-incompatible) version. Doxia 2.0 is the first version though which opens a migration path to a modern markup language (Markdown) through the newly added `MarkdownSink` which is only working properly in Doxia 2.0 with the newest converter. Therefore for me it is crucial to support the (original) FML/XDoc specification.

**Comment by jira-importer** [comment]
**[Michael Osipov](https://issues.apache.org/jira/secure/ViewProfile.jspa?name=michael-o)** commented

I'd like to know [Hervé Boutemy](https://issues.apache.org/jira/secure/ViewProfile.jspa?name=hboutemy)'s opinion. Both formats don't hurt and I know that Xdoc fully cover the parser and sink specs. It would be also awkward of their body content would diverge from the XHTML5 one.

**Comment by jira-importer** [comment]
**[Michael Osipov](https://issues.apache.org/jira/secure/ViewProfile.jspa?name=michael-o)** commented

Right and nothing stops us to bump major version on them. We should not maintain two XHTML parsers.

**Comment by jira-importer** [comment]
**[Hervé Boutemy](https://issues.apache.org/jira/secure/ViewProfile.jspa?name=hboutemy)** commented

FML is legacy, and even when not legacy, it was an edge case not used so much
xdoc is the XML view on the Sink API: by definition, to me, it cannot be legacy as it just translates the Sink API into an XML source syntax: if someone wants to use the pure Doxia API = the sink API, xdoc remains the way to go
ok, this use case remains an edge case, as nowadays people should prefer Markdown or asciidoc, but Markdown and asciidoc are a mix between Doxia Sink API supported features and raw HTML parsing / immediate rendering: having xdoc remains useful as a reminder (if my view still makes sense: over years, maybe I'm wrong now...)

in summary: ok to drop FML, not ok to drop xdoc

**Comment by jira-importer** [comment]
**[Michael Osipov](https://issues.apache.org/jira/secure/ViewProfile.jspa?name=michael-o)** commented

[Hervé Boutemy](https://issues.apache.org/jira/secure/ViewProfile.jspa?name=hboutemy), are you OK with uplifting XDoc content to XHML5 with XDoc 3.0? That would be just natural.

**Comment by jira-importer** [comment]
**[Michael Osipov](https://issues.apache.org/jira/secure/ViewProfile.jspa?name=michael-o)** commented

[Hervé Boutemy](https://issues.apache.org/jira/secure/ViewProfile.jspa?name=hboutemy), I do remember that you told me that there is no effort in keeping FML, and there is basically none. Now you think it is time to remove? If so, we should deprecate in this milestone and drop in the next one.

**Comment by jira-importer** [comment]
**[Konrad Windszus](https://issues.apache.org/jira/secure/ViewProfile.jspa?name=kwin)** commented

> XHML5 with XDoc 3.0? That would be just natural.

I disagree here, I don't think we should just drop XDoc 2 support (at least not without providing an automated transition path). Supporting it is not that hard. The differences are outlined in https://www.w3.org/TR/html5-diff/. I think writing an adapter from XHTML1 to XHTML5 for XDoc/FML is feasible.
